AWS EKS kubernetes resources

DaMandal0rian opened the pull request:
The PR adds terraform declarations for AWS EKS managed kubernetes clusters on AWS infrastructure. 

- Cluster and node pool resources are created
- IAM role and KMS key management
- Bottlerocket optimized cluster with custom AMI for better performance and security
- gitops with argocd
- enable coredns, prometheus, cert-manager addons

closes #143

vedhavyas commented on eks/base/eks_managed_node_group/main.tf:
```diff
@@ -0,0 +1,495 @@
+        lockdown = "integrity"
+
+        [settings.kubernetes.node-labels]
+        # todo: add label to the nodes
```
can we not get them through config and not hardcode it here ?

DaMandal0rian replied:
yeah done

vedhavyas commented on eks/base/eks_managed_node_group/variables.tf:
```diff
@@ -0,0 +1,84 @@
+  description = "List of AWS IAM accounts to add to aws-auth configmap"
+  default     = []
+
+  # i.e
```
do we need these commented out default ?

vedhavyas commented on eks/base/eks_managed_node_group/variables.tf:
```diff
@@ -0,0 +1,84 @@
+
+        [settings.kubernetes.node-labels]
+        # todo: add label to the nodes
+        app1 = "explorer-squid"
```
I still see these applications hard coded. Can we get these from config instead?

DaMandal0rian replied:
sure, I have refactored it now, created blue/green environments and added argocd too. These hardcoded values are passed as input variables now.

vedhavyas reviewed (commented):
Make sense. Left some suggestions

DaMandal0rian committed:
EKS managed kubernetes

add blue/green deployment and argocd

- use new eks module
- declarative deployments with argocd
- blue green envs
- prod env
- refactor userdata and remove hardcoded 
- network module and variables fix

DaMandal0rian committed:
change route53 resources and env variables
